### PR TITLE
Background jobs

### PR DESCRIPTION
**Pull request by squush**
May want to test this on your local computer - this was not working for me in the morning and the TAs couldn't figure out why, now I retry it and suddenly it works - I have a feeling it's more an issue with my computer than with the code itself.

### CODE REVIEW

File: app/jobs/scrape_job.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require 'nokogiri'
+require 'open-uri'
+require 'socket'
+
+class ScrapeJob < ApplicationJob
+  queue_as :default
+
+  def perform
```
Comment (andygw) [inline app/jobs/scrape_job.rb:8]: This looks like the older code. Was the newer code not working?

File: app/views/tournaments/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 <div class="container">
+  <h1 class="mb-3 border-bottom border-3"><%= @tournament.event.location %></h1>
```
Comment (andygw) [inline app/views/tournaments/show.html.erb:2]: Do we need all this stuff added back to the show page?

File: lib/tasks/batch.rake [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+require 'open-uri'
+
+namespace :batch do
+  desc "Scrape cross-tables.com for upcoming tournaments"
+  task scrape_xtables: :environment do
+    p "starting scrape"
+    url = 'https://www.cross-tables.com'
+    html = URI.open(url)
+    doc = Nokogiri::HTML(html)
+    tournaments = doc.css('#utblock .xtdatatable tr')
```
Comment (andygw) [inline lib/tasks/batch.rake:11]: This looks like the old code

**Review by andygw — CHANGES_REQUESTED** [review]
See inline comments

File: app/jobs/scrape_job.rb (hunk 1 above)
Comment (squush) [reply app/jobs/scrape_job.rb:8]: We no longer need the ScrapeJob, so I could remove this file

File: app/views/tournaments/show.html.erb (hunk 2 above)
Comment (squush) [reply app/views/tournaments/show.html.erb:2]: Seems okay, we need to display the location somewhere

File: lib/tasks/batch.rake (hunk 3 above)
Comment (squush) [reply lib/tasks/batch.rake]: All updated now

File: app/views/tournaments/show.html.erb (hunk 2 above)
Comment (andygw) [reply app/views/tournaments/show.html.erb:2]: It's already displayed in the h1 that was already there

File: lib/tasks/batch.rake [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+require 'open-uri'
+
+def create_event(event)
+  url = "https://www.cross-tables.com#{event.attribute('href').value}"
+  html = URI.open(url)
+  doc = Nokogiri::HTML(html)
+
+  # This returns a string date, since often the date is a range,
+  # which is not easy to parse into a Date object. See above
+  date = doc.css('p').children[2].text[/\w.*20\d\d/]
```
Comment (andygw) [inline lib/tasks/batch.rake:11]: Why do we define date here if the Event creation is using @sortable_date?

File: app/views/tournaments/show.html.erb (hunk 2 above)
Comment (squush) [reply app/views/tournaments/show.html.erb:2]: ah true

File: lib/tasks/batch.rake [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        @location = tournament.css('a').children.text
+
+        # May want to use this date method below to get the tournament's start
+        # date so that we can sort upcoming events by start date
+        # TODO: Fix the month_and_day regex to grab date ranges. Currently,
+        #       multi-day events are setting each event on the first day. Check
+        #       Kingston, as an example
+        month_and_day = tournament.css('td')[-2].children.text[/^\d*\/\d*/]
+        year = tournament.css('td')[-2].children.text[/\d{4}/] || Date.today.year
+        @sortable_date = Date.parse("#{year}/#{month_and_day}")
```
Comment (andygw) [inline lib/tasks/batch.rake:52]: I think we can comment this out. We should probably grab the date from the event page, not the table of events.

File: app/views/tournaments/show.html.erb [hunk 6]
```diff
@@ -1,5 +1,12 @@
 <div class="container">
-  <h1 class="mb-3 tourney-main"><%= @tournament.event.location %></h1>
+
```
Comment (andygw) [inline app/views/tournaments/show.html.erb:2]: What's going on here?

**Pull request merged**